scripts/un/energy/process.py

```diff
@@ -0,0 +1,560 @@
+    'populationType': 'dcs:Energy',
+    'statType': 'dcs:measuredValue',
+}
+
+UN_ENERGY_TMCF = """
+Node: E:UNEnergy->E0
+typeOf: dcs:StatVarObservation
+observationAbout: C:UNEnergy->Country_dcid
+variableMeasured: C:UNEnergy->StatVar
+observationDate: C:UNEnergy->Year
+observationPeriod: "P1Y"
+value: C:UNEnergy->Quantity
+unit: C:UNEnergy->Unit_dcid
+scalingFactor: C:UNEnergy->Scaling_factor
+measurementMethod: C:UNEnergy->Estimate
+"""
+
+
+def print_debug(debug_level: int, *args):
+    if debug_level > 1:
+        print("[", datetime.datetime.now(), "] ", *args, file=sys.stderr)
+
+
+def _print_counters(counters, steps=None):
+    row_key = 'inputs_processed'
+    if steps is None or row_key not in counters or counters[
+            row_key] % steps == 0:
+        print('\nSTATS:')
+        for k in sorted(counters):
+            print(f"\t{k} = {counters[k]}")
+        if 'inputs_processed' in counters:
+            start_ts = counters['time_start']
+            end_ts = time.perf_counter()
+            print(
+                'Processing rate: {:.2f}'.format(counters['inputs_processed'] /
+                                                 (end_ts - start_ts)),
+                'rows/sec')
+        print('', flush=True)
+
+
+def _add_error_counter(counter_name: str, error_msg: str, counters):
+    print_debug(2, "Error: ", counter_name, error_msg)
+    if counters is not None:
+        debug_lines = 1
+        if 'debug_lines' in counters:
+            debug_lines = counters['debug_lines']
+        if counters[counter_name] % debug_lines == 0:
+            print("ERROR: ", counter_name, ": ", error_msg)
+        counters[counter_name] += 1
+
+
+def remove_extra_characters(name: str) -> str:
+    """Removes the parts of the name that is not used in the node id,
+    including:
+         - any namespace: prefix, such as 'dcs:' or 'dcid:'
+         - capitalized prefix of two or more letters
+         - Any '_'
+    For example: 'dcid:EIA_Other_fuel' will be converted to: 'OtherFuel'
+
+    Args:
+      name: string to be normalized.
+
+    Returns:
+      string without the extra characters and capitalized appropriately.
+    """
+    if name is None:
+        return name
+    # strip namespace: prefix
+    name = name[name.find(':') + 1:]
+    # string any prefix of 2 or more upper case letters.
+    upper_prefix = 0
+    while upper_prefix < len(name):
+        if not name[upper_prefix].isupper():
+            break
+        upper_prefix += 1
+    if upper_prefix > 1:
+        name = name[upper_prefix:]
+        if name[0] == '_':
+            name = name[1:]
+    # Replace all '_' with a capitalized letter.
+    # Find all occurences of '_'.
+    upper_idx = [-1] + [i for i, e in enumerate(name) if e == '_'] + [len(name)]
+    # Capitalize the next letter after '_'.
+    words = [
+        name[x + 1].upper() + name[x + 2:y]
+        for x, y in zip(upper_idx, upper_idx[1:])
+        if name[x:y] != '_'
+    ]
+    return ''.join(words)
```

Comment:
This function and the ones above seem reusable...

Reply:
Should i move them to a common data/util, say 'counters.py' and 'dcid_utils.py'?

Q: How do I import from other parts of the data tree?
  It seems like the script needs to be run from root to allow such imports:
  from util import counters

Reply:
Feels like this the latter might have overlap with https://github.com/datacommonsorg/data/pull/489/files.

Option 1: https://github.com/datacommonsorg/data/blob/master/scripts/india_plfs/daily_wage_data/preprocess.py#L25
 (implication is we can only run this script from scripts/)
Option 2: https://github.com/datacommonsorg/data/blob/cdf4c6532be3e55dc91b621bf2b28ee533ad09ac/scripts/us_eia/eia_860/utils.py#L25-L26 (weirder but more flexible I think).

This also ties in with the discussion of structuring the py modules/tests in a consistent way in the whole repo.